ui/src/main/js/page/task/TaskModal.js

```diff
@@ -50,23 +50,16 @@ const TaskModal = ({ data, isOpen, toggleModal }) => {
                     value={nextRunTime}
                 />
 
-                {/* If properties array has content, Provider will be present in first position */}
-                { properties[0] ? (
-                    <ReadOnlyField
-                        id={setId('provider')}
-                        label="Provider"
-                        value={properties[0].value}
-                    />
-                ) : null }
 
-                {/* If properties array has content, Configuration will be present in second position */}
-                { properties[1] ? (
-                    <ReadOnlyField
-                        id={setId('configuration')}
-                        label="Configuration"
-                        value={properties[1].value}
-                    />
-                ) : null }
+                {/* If properties array has content, Provider will be present in first position */}
+                { properties && properties.map(property => (
+                        <ReadOnlyField
+                            id={setId(property.key)}
+                            label={property.displayName}
+                            value={property.value}
+                        />
+                    ))
+                }
```

Comment:
What is this change for?

Reply:
Tasks has a collection of metadata properties that are set on the back end.  We always assumed if we had 2 properties it was provider UUID and then the provider configuration.  I made the UI be dynamic and add fields based on the properties passed from the back end.  This works for Provider configuration and for the new tasks I have added.